### PR TITLE
Replace yapf with black & isort

### PR DESCRIPTION
**Pull request by bayandin**
TL;DR:
- `yapf` is replaced with `black` and `isort`, code autoformatted
- `flake8` added to pre-commit hook, and all warnings are fixed manually in [this commit](https://github.com/neondatabase/neon/pull/2297/commits/6d933a005637f32909535d8e580a0e84b2180442)
- Tools config moved from `setup.cfg` to `pyproject.toml` (all but `flake8` which doesn't support `pyproject.toml` yet)

---
My initial issue with `yapf` was that it doesn’t support some modern python syntax, like [parenthesized context managers](https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers):
```python
with (
    open('f1') as f1,
    open('f2') as f2,
):
    pass
```

And `yapf` can’t force quotes style (https://github.com/google/yapf/issues/399).

Despite the mentioned issues, it boils down to personal preferences; I believe we can have a more unified code style with black.

A note to myself: do not squash for the merge otherwise, adding `.git-blame-ignore-revs` in the next PR will be impossible.

### CODE REVIEW

**Comment by hlinnaka** [comment]
Why? No objections, and I don't have an opinion on which formatter does a better job, but I'm curious if there was a reason to switch.

**Comment by bayandin** [comment]
> Why? No objections, and I don't have an opinion on which formatter does a better job, but I'm curious if there was a reason to switch.

Added to the PR description

File: test_runner/fixtures/neon_fixtures.py [hunk 1]
```diff
@@ -1049,17 +1073,19 @@ def timeline_detail(self,
             include_non_incremental_physical_size_str = "1"
 
         res = self.get(
-            f"http://localhost:{self.port}/v1/tenant/{tenant_id.hex}/timeline/{timeline_id.hex}" +
-            "?include-non-incremental-logical-size={include_non_incremental_logical_size_str}" +
-            "&include-non-incremental-physical-size={include_non_incremental_physical_size_str}")
+            f"http://localhost:{self.port}/v1/tenant/{tenant_id.hex}/timeline/{timeline_id.hex}"
+            + f"?include-non-incremental-logical-size={include_non_incremental_logical_size_str}"
+            + f"&include-non-incremental-physical-size={include_non_incremental_physical_size_str}"
```
Comment (bayandin) [inline test_runner/fixtures/neon_fixtures.py:1078]: This was a bug found by `flake8`

File: test_runner/fixtures/neon_fixtures.py [hunk 2]
```diff
@@ -1049,17 +1073,19 @@ def timeline_detail(self,
...
+        )
```
Comment (bayandin) [inline test_runner/fixtures/neon_fixtures.py:1079]: This was a real bug found by `flake8`: we didn't use f-string for `include-non-incremental-physical-size` and `include-non-incremental-logical-size` GET parameters

Comment (LizardWizzard) [reply test_runner/fixtures/neon_fixtures.py:1079]: This can be streamlined with something like that:

```python
params = {}
if include_non_incremental_logical_size:
    params['include-non-incremental-logical-size'] = 'present'

if include_non_incremental_physical_size:
    params['include-non-incremental-physical-size'] = 'present'

res = self.get(
    f"http://localhost:{self.port}/v1/tenant/{tenant_id.hex}/timeline/{timeline_id.hex}",
    params=params,
)
```

Rust code checks only for parameter presence to avoid dealing with all possible truthy variants like "y, yes, 1" etc

**Comment by bayandin** [comment]
> Personally I prefer single quotes when it is possible (with the exception when single quotes are located inside the literal).
black has --skip-string-normalization option, which will prevent it from converting everything from double quotes. I'm not insisting on enabling it but asking for opinion :) 

I don't have a strong opinion about which type of quotes to use, but I'd like to have them consistent. And since black enforces only double quotes, so let it be double quotes. 
Regarding `--skip-string-normalization`, do you mean to use `--skip-string-normalization` for this PR and not use it for the following to let quotes be unified eventually? 
I'd prefer to have everything change in a single commit than spread over time. Seeing both types of quotes in the same file is confusing (especially for new joiners).

**Review by LizardWizzard — APPROVED** [review]
> do you mean to use --skip-string-normalization for this PR and not use it for the following to let quotes be unified eventually?

I'm for setting it permanently in config and completely avoiding large portion of corresponding changes.

> Seeing both types of quotes in the same file is confusing (especially for new joiners).

Do not find it confusing. Maybe because I'm used to it. There will be always situations when you need both types of quotes to avoid ugly escaping sequences (thats again quite subjective and other oepple may have different opinion). So knowing that they have equal meaning is sort of needed anyway. Single quotes are used in official python documentation. There is a long [thread](https://github.com/psf/black/issues/118#issuecomment-379898677) in black which lead to adding --skip-string-normalization and there was quite a lot of debate around that. 

There are some common styles. Some people say that for messages displayed to users you should use double quotes, and for constants you should use single quotes. https://stackoverflow.com/a/56190 Another one is to use single quotes whenever possible https://stackoverflow.com/a/56190. I'm from the second camp. For me the reason is that single quote requires one key stroke (which may be not true for certain keyboard layouts, but it is true for most common EN_US QWERTY one) and because single quotes has the same meaning why bother and press extra key? With black it is somewhat simpler, by default it will just rewrite everything with double quotes at pre-commit stage. So I'm not insisting. Do not want to start another tabs vs spaces battle :) So I'm fine with either option. Up to you.
